### ISSUE TITLE
CoreML Sampling Crash In `swift-transformers`

### ISSUE THREAD

**Issue by SpiraMira** [huggingface/swift-transformers#365]
## Summary

When using `swift-transformers` with a local Core ML language model, greedy decoding is stable, but sampling-based decoding crashes the process in Apple native code (`libBNNS.dylib`) instead of throwing a recoverable Swift error.

This happens during token selection inside `selectNextTokenUsingSampling(from:)`.

## Environment & System Under Test (SUT)

- **Device/Hardware**: Apple Silicon Mac (M-series) / iPad
- **OS Version**: macOS Tahoe 26.5 (Build 25F71) / iOS 18+
- **Xcode Version**: Xcode 26.5 (Build 17F42)
- **Target SDKs**: macOS 14.4+ / iOS 17.0+
- **Swift Compiler**: Swift 6.0 (Strict Concurrency: Complete)
- **Model Runtime**: Core ML local model loaded through `Models.LanguageModel.loadCompiled(...)`
- **App Integration**: Swift client application wrapping compiled Core ML models directly
- **Model Characteristics**:
  - causal LM
  - rank-3 logits
  - large vocab (`151936`)
  - local Core ML model, not remote inference

## Runtime Dependency & Tooling Versions

- **`swift-transformers`**: 1.1.6 (huggingface package revision: `573e5c9`)
- **Hugging Face `AnyLanguageModel`**: 0.8.0 (revision: `163f385`)
- **Python Environment**: Python 3.10.16 
- **`coremltools` (Converter)**: 7.1 (compiled with PyTorch 2.2.1)

## Development Dependency & Tooling Versions for model conversion 
- **`python`**: 3.11.15
- **`coremltools`** : 9.0
- **`transformers`** : 5.8.1
- **`tokenizers`** : 0.22.2
- **`torch`** : 2.7.0

## Observed Behavior

Greedy decoding works.

Sampling mode crashes the process with `EXC_BAD_ACCESS` in `libBNNS.dylib`.

Relevant stack shape:

- `Generation.generate(...)`
- `selectNextTokenUsingSampling(from:)` at `Sources/Generation/Decoders.swift:47`
- then crash in `libBNNS.dylib`

The crash is native and bypasses Swift error handling, so callers cannot catch it or recover.

## Expected Behavior

Sampling should either:

1. work correctly, or
2. fail with a recoverable Swift error

It should not hard-crash the process.

## Suspected Trigger

The crash appears specific to the Core ML sampling path, not model load:

- the model loads successfully
- generation starts
- crash occurs during sampling token selection

The most suspicious code path is the Core ML tensor sampling implementation in:

- `selectNextTokenUsingSampling(from:)`

which does:

- `softmax`
- `cumulativeSum`
- random tensor generation
- comparison mask
- arithmetic over the full vocab tensor
- `argmin`

Because greedy decoding is stable on the same model, this seems isolated to sampling rather than the general Core ML generation path.

## Minimal Behavioral Difference

Stable:

- `doSample = false`

Crashes:

- any configuration that results in `generationMode == .sample`
- for example, top-p sampling enabled

## Workaround

We are currently forcing deterministic decoding (`doSample = false`) for local Core ML models to avoid the crash.

## Additional Note

This was observed on a model with a very large vocab (`151936`), so tensor size may be relevant.

## Representative Call Chain

- ClientApplication.runGenerationTask(...)
- Generation.generate(...)
- selectNextTokenUsingSampling(from:)

## Crash Signature

Representative stop reason:

- `EXC_BAD_ACCESS`
- top native frame in `libBNNS.dylib`

Representative stack shape:

```text
Generation.generate(...)
selectNextTokenUsingSampling(scores=...) at Decoders.swift:47
libBNNS.dylib
```

## Why This Is High Impact

The crash happens in a normal text-generation path for local Core ML models and is not catchable in Swift. Any app enabling sampling for local Core ML generation can lose the entire process rather than surfacing a model/runtime error.
